packages/spaces/src/SpacesLink.js

```diff
@@ -0,0 +1,318 @@
+import React, { useMemo, useCallback, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import truncate from 'lodash.truncate';
+import ReactMarkdown from 'react-markdown';
+import { FavoriteHeart } from '@availity/favorites';
+import { Card, CardBody, Media, CardText, CardTitle, Badge, ListGroupItemHeading, ListGroupItemText } from 'reactstrap';
+import ListGroupItem from '@availity/list-group-item';
+import dayjs from 'dayjs';
+import AppIcon from '@availity/app-icon';
+import Icon from '@availity/icon';
+import { useSpacesContext } from './Spaces';
+import { isFunction } from './helpers';
+import useLink from './useLink';
+import Loader, { skeletonPropType } from './Loader';
+import '../styles.scss';
+
+const getDisplayDate = (date) => dayjs(date).format('MM/DD/YYYY');
+
+const getContainerTag = (propTag = 'div', linkStyle) =>
+  ({
+    card: Card,
+    list: ListGroupItem,
+  }[linkStyle] || propTag);
+
+const getBodyTag = (propTag = 'div', linkStyle) =>
+  ({
+    card: CardBody,
+    list: 'div',
+  }[linkStyle] || propTag);
+
+const getTitleTag = (propTag, linkStyle) =>
+  propTag ||
+  {
+    card: CardTitle,
+    list: ListGroupItemHeading,
+  }[linkStyle] ||
+  'div';
+
+const getTextTag = (propTag = 'div', linkStyle) =>
+  ({
+    card: CardText,
+    list: ListGroupItemText,
+  }[linkStyle] || propTag);
+
+const Link = ({
+  spaceId,
+  space: propSpace,
+  className,
+  children,
+  appIcon: showAppIcon,
+  favorite,
+  icon,
+  showNew,
+  showDate,
+  stacked,
+  body,
+  description: showDescription,
+  tag: Tag,
+  bodyTag: BodyTag,
+  titleTag: TitleTag,
+  textTag: TextTag,
+  linkStyle,
+  size,
+  loading: propsLoading,
+  clientId: propsClientId,
+  maxDescriptionLength, // TODO: remove and replace with text-truncate css
+  style,
+  skeletonProps,
+  linkAttributes,
+  role,
+  analytics,
+  ...rest
+}) => {
+  const { loading } = useSpacesContext() || {};
+  const isLoading = loading || propsLoading;
+
+  const [
+    {
+      id,
+      name,
+      shortName,
+      type,
+      metadata,
+      description,
+      activeDate,
+      isNew,
+      icons = {},
+      images = {},
+      colors = {},
+      link,
+      ...restLink
+    } = {},
+    props = {},
+  ] = useLink(propSpace || spaceId, {
+    clientId: propsClientId,
+    linkAttributes,
+  });
+
+  const getIconTitle = useCallback(() => {
+    if (shortName) return shortName;
+
+    // We have to pass `name` as `className` bc of how its stored in spaces
+    if (icons.navigation)
+      return (
+        <Icon className={icons.navigation} data-testid={`space-link-${icons.navigation}-icon-${linkStyle}-${id}`} />
+      );
+
+    return <Icon name="desktop" data-testid={`space-link-desktop-icon-${linkStyle}-${id}`} />;
+  }, [icons.navigation, shortName]);
```

Comment:
Any specific reason to not add id and linkStyle? I think we should have them in the dep array

Reply:
They're only used in the data-testid, so that would cause unnecessary re-renders in the deployed version, right?